### ISSUE TITLE
simulation breaks down when n_instances parameter is set too high

### ISSUE THREAD

**Issue by GerbrichFerdinands** [asreview/asreview#662]
**Describe the bug**
The `--n_instances` parameter (number of records queried each query) has a ceiling, where the simulation breaks. 

**To Reproduce**
I use the the PTSD dataset (n = 6189), with 5 prior inclusions and 5 prior exclusions, and with `--n_instances` set to 6179 (6189-10, # of records not used for training). 

**Screenshots**

```
gerbrich asreview simulate van_de_Schoot_2017.csv --n_prior_included 5 --n_prior_excluded 5 --n_instances 6179 --state_file default_10prior_passive.h5 --seed 42 --init_seed 42

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

The following records are prior knowledge:

4939 - Posttraumatic stress disorder symptom trajectories in Hurricane Katrina affect..   Self-Brown, S., Lai, B. S., Thompson, ..
3483 - Trajectories of Scores on a Screening Instrument for PTSD Among World Trade Ce..   Maslow, C. B., Caramanica, K., Welch, ..
3489 - Growth curve trajectories of distress in burn patients                             Mason, S. T., Corry, N., Gould, N. F.,..
4938 - Maternal posttraumatic stress disorder symptom trajectories following Hurrican..   Self-Brown, S., Lai, B. S., Harbin, S...
4435 - Trajectories of posttraumatic stress symptoms (PTSS) after major war among Pal..   Punamäki, R. L., Palosaari, E., Diab, ..
4403 - New approaches to combining pharmacotherapy and psychotherapy for posttraumati..   Pratchett, L. C., Daly, K., Bierer, L...
3496 - Impact of baseline mental health on adherence to interferon-free HCV therapy       Masur, J., Tang, L., Nelson, A., Osinu..
4991 - Profiles of connectedness: Processes of resilience and growth in children with..   Sharp, K. M. H., Willard, V. W., Okado..
1004 - Headache diagnoses among iraq and afghanistan war veterans enrolled in VA: A g..   Carlson, K. F., Taylor, B. C., Hagel, ..
2785 - Clinical Evaluation of a Measure to Assess Combat Exposure                         Keane, T. M., Fairbank, J. A., Caddell..
Traceback (most recent call last):
  File "/Users/gerbrich/asreview/bin/asreview", line 8, in <module>
    sys.exit(main())
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/__main__.py", line 80, in main
    entry.load()().execute(sys.argv[2:])
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/entry_points/simulate.py", line 47, in execute
    review_simulate(path, **args_dict)
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/review/factory.py", line 327, in review_simulate
    review(dataset, *args, mode='simulate', **kwargs)
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/review/factory.py", line 290, in review
    reviewer.review()
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/review/base.py", line 336, in review
    self._do_review(state, *args, **kwargs)
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/review/base.py", line 301, in _do_review
    self.log_current_query(state)
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/review/base.py", line 356, in log_current_query
    state.set_current_queries(self.shared["current_queries"])
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/asreview/state/hdf5.py", line 77, in set_current_queries
    self.f.attrs["current_queries"] = data
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/h5py/_hl/attrs.py", line 100, in __setitem__
    self.create(name, data=value)
  File "/Users/gerbrich/asreview/lib/python3.8/site-packages/h5py/_hl/attrs.py", line 201, in create
    attr = h5a.create(self._id, self._e(tempname), htype, space)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5a.pyx", line 47, in h5py.h5a.create
RuntimeError: Unable to create attribute (object header message is too large)
```

I think asreview being unable to handle such large `--n_instances` values might
have something to do with the current design of the state file. I tested it
with `--n_instances` 10, 100, 1000, 3000, and 4000 as well, which works fine.
When I tried at 6000 (or higher), it fails. With a smaller dataset (n = around 1000) I had no problems whatsoever. 

**Version information**
 - OS: macOS
 - ASReview version 0.17

**Additional context**
I want to run a model that trains only once, e.g. only on the prior records. I try to arrange for that by making sure the first query consists of *all* the records in the data minus the prior records.

**Comment by J535D165** [comment]
Thanks for reporting. Interesting issue (and nice application ;) ). 

We might want to implement a logic over here: 
https://github.com/asreview/asreview/blob/2bf7459f93cb0e7bbb9d7d1dcafcf4e15e23d5f3/asreview/review/base.py#L111-L169. 

I can think about two potential solutions: raise an informative error or slice `n_instances` to the maximum value possible. 

While writing this, I think the logic can be challenging, because we also have `n_papers` and `n_queries`. This might complicate things. The logic should be a combination of these four parameters: 

https://github.com/asreview/asreview/blob/2bf7459f93cb0e7bbb9d7d1dcafcf4e15e23d5f3/asreview/review/base.py#L93-L104

**Comment by J535D165** [comment]
I think I misunderstood the issue. There is a problem with allocating memory in the attributes of the state file for large queries like this. Hmmm. 

> We generally consider the maximum size of an attribute to be 64K bytes. The library has two ways of storing attributes larger than 64K bytes: in dense attribute storage or in a separate dataset. Using dense attribute storage is described in this section, and storing in a separate dataset is described in the next section.
> 
> To use dense attribute storage to store large attributes, set the number of attributes that will be stored in compact storage to 0 with the H5Pset_attr_phase_change function. This will force all attributes to be put into dense attribute storage and will avoid the 64KB size limitation for a single attribute in compact attribute storage.
http://davis.lbl.gov/Manuals/HDF5-1.8.7/UG/13_Attributes.html

I can't think about an easy fix at the moment. The new state file will fix this issue for sure.

**Comment by PeterLombaers** [comment]
It tries to log something called `current_queries` in a part of the hdf5 file which is not allowed to be too big. Normally that is not a problem, but apparently when you have too many papers at the same time, this goes wrong. I think you can get a very hacky temporary solution by commenting out line 301: `self.log_current_query(state)` in asreview.review.base.py. And I think that only works because you don't need to train the model after the first time.

Alternatively you could try using the Python package sklearn directly, but it would be a bit of a hassle to get all the settings the same. With the new state file this definitely won't be an issue.

**Comment by J535D165** [comment]
Resolved in #688
